packages/service-clients/odsp-client/src/odspContainerServices.ts

```diff
@@ -3,25 +3,71 @@
  * Licensed under the MIT License.
  */
 
+import { TypedEventEmitter } from "@fluid-internal/client-utils";
 import type { IContainer } from "@fluidframework/container-definitions/internal";
+import type { IDisposable } from "@fluidframework/core-interfaces";
 import { createServiceAudience } from "@fluidframework/fluid-static/internal";
 
 import type {
 	IOdspAudience,
 	OdspContainerServices as IOdspContainerServices,
+	IOdspContainerServicesEvents,
 } from "./interfaces.js";
 import { createOdspAudienceMember } from "./odspAudience.js";
 
 /**
  * @internal
  */
-export class OdspContainerServices implements IOdspContainerServices {
+export class OdspContainerServices
+	extends TypedEventEmitter<IOdspContainerServicesEvents>
```

Comment:
Ah sorry, just saw this was new usage.  For new usage it's preferable to have an `events` member rather than class inheritance, and use `createEmitter` instead of `TypedEventEmitter`.  Something like

```ts
	readonly #events = createEmitter<IOdspContainerServicesEvents>();
	public get events(): Listenable<IOdspContainerServicesEvents> {
		return this.#events;
	}
```

The event interface changes a bit too, something like:
```ts
export interface IOdspContainerServicesEvents {
	readOnlyStateChanged: () => void;
	sensitivityLabelsInfoChanged: () => void;
}
```